compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -168,11 +167,8 @@ pub fn check_crate(tcx: TyCtxt<'_>) {
         let def_kind = tcx.def_kind(item_def_id);
         match def_kind {
             DefKind::Static { .. } => tcx.ensure().eval_static_initializer(item_def_id),
-            DefKind::Const if tcx.generics_of(item_def_id).is_empty() => {
-                let instance = ty::Instance::new(item_def_id.into(), ty::GenericArgs::empty());
-                let cid = GlobalId { instance, promoted: None };
-                let typing_env = ty::TypingEnv::fully_monomorphized();
-                tcx.ensure().eval_to_const_value_raw(typing_env.as_query_input(cid));
+            DefKind::Const if !tcx.generics_of(item_def_id).requires_monomorphization(tcx) => {
+                tcx.ensure().const_eval_poly(item_def_id.into())
```

Comment:
This temporarily 'reverts' the perf optimization from #121387. I have to see what to do about this.

Reply:
IINM `const_eval_poly` would be the 'most correct' option since we do want to construct the `Instance` with identity substs (contrary to empty ones) and likely do want use the ParamEnv of the free constant lol (tho I've yet find an example where the latter makes a difference it practice)